geomesa-tools/src/main/scala/org/locationtech/geomesa/tools/commands/ExportCommand.scala

```diff
@@ -55,18 +55,17 @@ class ExportCommand(parent: JCommander) extends CommandWithCatalog(parent) with
   }
 
   def getFeatureCollection(fmt: String): SimpleFeatureCollection = {
+    val sft = ds.getSchema(params.featureName)
     fmt match {
       case SHP =>
         val schemaString =
           if (Option(params.attributes).nonEmpty) {
-            params.attributes
+            ShapefileExport.replaceGeomInAttributesString(params.attributes, sft)
           } else {
-            val sft = ds.getSchema(params.featureName)
             ShapefileExport.modifySchema(sft)
```

Comment:
I think this method is broken as well - did you test without passing any attributes?